internal/nodes/avi_model_evh_nodes.go

```diff
@@ -1375,19 +1474,6 @@ func RemoveRedirectHTTPPolicyInModelForEvh(vsNode *AviEvhVsNode, hostnames []str
 	}
 }
 
-func RemoveFQDNsFromModelForEvh(vsNode *AviEvhVsNode, hosts []string, key string) {
-	if len(vsNode.VSVIPRefs) > 0 {
-		for i, fqdn := range vsNode.VSVIPRefs[0].FQDNs {
-			if utils.HasElem(hosts, fqdn) {
-				// remove logic conainer-lib candidate
-				vsNode.VSVIPRefs[0].FQDNs[i] = vsNode.VSVIPRefs[0].FQDNs[len(vsNode.VSVIPRefs[0].FQDNs)-1]
```

Comment:
The removal logic was not working if the hosts contains more than one value.

Refer here for a [sample program](https://go.dev/play/p/yAZGDW5ADHA) illustrating it.